### ISSUE TITLE
Multi region errors

### ISSUE THREAD

**Issue by rgontarek** [terraform-aws-modules/terraform-aws-transit-gateway#14]
Does this module support multi account vpc associations and if yes, what is the proper way to pass the provider to the calling resource?

```
provider "aws" {
  region = "us-east-1"
  alias    = "one"
}
provider "aws" {
  region = "us-east-1"
  alias    = "two"
}

// See Notes in README.md for explanation regarding using data-sources and computed values
data "aws_vpc" "default" {
  default = true
}

data "aws_subnet_ids" "this" {
  vpc_id = data.aws_vpc.default.id
}

module "tgw" {
  **providers = {aws = "aws.one"}**
  source = "../../"
  name            = "my-tgw"
  description     = "My TGW shared with several other AWS accounts"
  amazon_side_asn = 64532
  enable_auto_accept_shared_attachments = true
  vpc_attachments = {
    vpc1 = {
      **providers = {aws = "aws.one"}**
      vpc_id                                          = data.aws_vpc.default.id      # module.vpc1.vpc_id
      subnet_ids                                      = data.aws_subnet_ids.this.ids # module.vpc1.private_subnets
      dns_support                                     = true
      ipv6_support                                    = true
      transit_gateway_default_route_table_association = false
      transit_gateway_default_route_table_propagation = false

      tgw_routes = [
        {
          destination_cidr_block = "30.0.0.0/16"
        },
        {
          blackhole              = true
          destination_cidr_block = "0.0.0.0/0"
        }
      ]
    },
    vpc2 = {
      **providers = {aws = "aws.two"}**
      vpc_id     = data.aws_vpc.default.id      # module.vpc2.vpc_id
      subnet_ids = data.aws_subnet_ids.this.ids # module.vpc2.private_subnets

      tgw_routes = [
        {
          destination_cidr_block = "50.0.0.0/16"
        },
        {
          blackhole              = true
          destination_cidr_block = "10.10.10.10/32"
        }
      ]
    },
  }

  ram_allow_external_principals = false
  ram_principals                = [************]

  tags = {
    Purpose = "tgw-complete-example"
  }
}

module "vpc1" {
  providers = {aws = "aws.one"}
  source  = "terraform-aws-modules/vpc/aws"
  version = "~> 2.0"
  name = "vpc1"
  cidr = "10.10.0.0/16"
  azs             = ["eu-west-1a", "eu-west-1b", "eu-west-1c"]
  private_subnets = ["10.10.1.0/24", "10.10.2.0/24", "10.10.3.0/24"]
  enable_ipv6                                    = true
  private_subnet_assign_ipv6_address_on_creation = true
  private_subnet_ipv6_prefixes                   = [0, 1, 2]
}

module "vpc2" {
  providers = {aws = "aws.two"}
  source  = "terraform-aws-modules/vpc/aws"
  version = "~> 2.0"
  name = "vpc2"
  cidr = "10.20.0.0/16"
  azs             = ["eu-west-1a", "eu-west-1b", "eu-west-1c"]
  private_subnets = ["10.20.1.0/24", "10.20.2.0/24", "10.20.3.0/24"]
  enable_ipv6 = false
}

```

**Comment by frednotet** [comment]
This version doesn't support the multi-account right now. I did some tests on a fork, but my provider need to be dynamic, which is the main issue that blocks me to get a working version... damned :(

Does someone know a workaround for this: `provider = each.value["provider"]` ?

**Comment by ChefAustin** [comment]
Do you mean “multi-region” instead of “multi-account”?

**Comment by antonbabenko** [comment]
Fixed in #20, and [v1.3.0](https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/releases/v1.3.0) has been just released.

**Comment by nibatandukar** [comment]
Hi @rgontarek , how did you solve this issue. I have got the same issue for multi region in the transit gateway. Can you please guide me?
